app/frontend/src/features/communities/hooks.ts

```diff
@@ -126,23 +126,24 @@ export const useListNearbyUsers = (communityId?: number) =>
     }
   );
 
-export const useNewDiscussionMutation = (queryClient: QueryClient) =>
-  useMutation<
-    Discussion.AsObject,
-    GrpcError,
-    {
-      title: string;
-      content: string;
-      ownerCommunityId: number;
-    }
-  >(
+export interface CreateDiscussionInput {
+  title: string;
+  content: string;
+  ownerCommunityId: number;
+}
+
+export const useNewDiscussionMutation = (onSuccess?: () => void) => {
+  const queryClient = useQueryClient();
+  return useMutation<Discussion.AsObject, GrpcError, CreateDiscussionInput>(
     ({ title, content, ownerCommunityId }) =>
       service.discussions.createDiscussion(title, content, ownerCommunityId),
     {
       onSuccess(_, { ownerCommunityId }) {
+        onSuccess?.();
         queryClient.invalidateQueries(
```

Comment:
Haven't a lot of time to test this yet to see if it's a problem, but since this is an infinite query, I wonder if we get any problems if we do a "naive" optimistic update and put an extra element in the latest/last page when it should have been the next page.

Otherwise the logic might be a bit more involved before and requires checking for page boundaries etc which could get a bit fiddly. Will have to experiment and see I guess...

Reply:
^ This is exactly as I feared and is non-trivial to do an optimistic update as soon as there are more than one page of discussions for a community. This is because each new discussion will need to correctly put the new discussion at the beginning, move the last discussion on the first page to the second page and repeat this array shunting. And since the `discussionList` itself is nested within an object for each `pages`, it's a bit of a pain to do and not quite as simple as merging arrays together...

Not sure if it's worth the effort and it might be easier to put up a "fetch" loading state on this first pass. I might still try write something up on the next PR part.

Reply:
Yeah I think that's Fine, optimistic updated aren't essential, it's more for the leading state do using `isFetching` is probably enough